tests/matrix_free/matrix_vector_26.cc

```diff
@@ -0,0 +1,271 @@
+  if (tria.begin(tria.n_levels() - 1)->is_locally_owned())
+    tria.begin(tria.n_levels() - 1)->set_refine_flag();
+  if (tria.last()->is_locally_owned())
+    tria.last()->set_refine_flag();
+  tria.execute_coarsening_and_refinement();
+  cell = tria.begin_active();
+  for (unsigned int i = 0; i < 10 - 3 * dim; ++i)
+    {
+      cell                 = tria.begin_active();
+      unsigned int counter = 0;
+      for (; cell != endc; ++cell, ++counter)
+        if (cell->is_locally_owned())
+          if (counter % (7 - i) == 0)
+            cell->set_refine_flag();
+      tria.execute_coarsening_and_refinement();
+    }
+
+  FE_Q<dim>       fe(fe_degree);
+  DoFHandler<dim> dof(tria);
+  dof.distribute_dofs(fe);
+
+  IndexSet owned_set = dof.locally_owned_dofs();
+  IndexSet relevant_set;
+  DoFTools::extract_locally_relevant_dofs(dof, relevant_set);
+
+  AffineConstraints<double> constraints(relevant_set);
+  DoFTools::make_hanging_node_constraints(dof, constraints);
+  VectorTools::interpolate_boundary_values(dof,
+                                           0,
+                                           Functions::ZeroFunction<dim>(),
+                                           constraints);
+  constraints.close();
+
+  deallog << "Testing " << dof.get_fe().get_name() << std::endl;
+  // std::cout << "Number of cells: " << tria.n_global_active_cells() <<
+  // std::endl; std::cout << "Number of degrees of freedom: " << dof.n_dofs() <<
+  // std::endl; std::cout << "Number of constraints: " <<
+  // constraints.n_constraints() << std::endl;
+
+  MatrixFree<dim, number> mf_data;
+  {
+    const QGauss<1>                                  quad(fe_degree + 1);
+    typename MatrixFree<dim, number>::AdditionalData data;
+    data.tasks_parallel_scheme = MatrixFree<dim, number>::AdditionalData::none;
+    data.tasks_block_size      = 7;
+    data.store_ghost_cells     = true;
+    mf_data.reinit(MappingQ1<dim>{}, dof, constraints, quad, data);
+  }
+
+  // Check if every ghosted cell is included in MatrixFree storage
+  {
+    using IndexLevel = std::pair<unsigned int, unsigned int>;
+    std::set<std::pair<unsigned int, unsigned int>> ghost_cells;
+
+
+
+    for (unsigned int batch = 0;
+         batch < mf_data.n_cell_batches() + mf_data.n_ghost_cell_batches();
+         ++batch)
+      {
+        for (unsigned int lane = 0;
+             lane < mf_data.n_active_entries_per_cell_batch(batch);
+             ++lane)
+          {
+            const typename Triangulation<dim>::active_cell_iterator cell =
+              mf_data.get_cell_iterator(batch, lane);
+            IndexLevel index_level(cell->index(), cell->level());
+            ghost_cells.insert(index_level);
+          }
+      }
+
+    for (const auto &cell : tria.active_cell_iterators())
+      if (cell->is_ghost())
+        {
+          IndexLevel index_level(cell->index(), cell->level());
+
+          AssertThrow(ghost_cells.find(index_level) != ghost_cells.end(),
```

Comment:
This only checks if the `MatrixFree` object can loop over all ghost cells. But does this also mean that you can access all necessary data there (e.g., mapping, indices)?